src/main/java/com/anonymous/privacytool/service/UserService.java

```diff
@@ -1,5 +1,18 @@
 package com.anonymous.privacytool.service;
 
+import com.anonymous.privacytool.dto.GenericResponse;
+import com.anonymous.privacytool.entity.User;
+import org.springframework.http.ResponseEntity;
+
+import java.util.Optional;
+
 public interface UserService {
     void processForgotPassword(String email);
+
+    ResponseEntity<GenericResponse> signUpUser(User user) throws Exception;
+
+    Optional<User> getUser(String  email);
+
+    ResponseEntity<GenericResponse> verifyUserLogin(User user) throws Exception;
```

Comment:
Prefer custom exceptions..
We can map error responses.
Improved error handling. i.e. Database error/Runtime Error - Server crashed - all get caught up in Exception.